src/App.tsx

```diff
@@ -1,222 +1,212 @@
   const user = useContext(AuthContext);
   const newTodoField = useRef<HTMLInputElement>(null);
+  const [newTitleTodo, setNewTitleTodo] = useState('');
+
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [, setTodoId] = useState(0);
+
+  const [changTitle, setChangTitle] = useState('');
+  const [statusFilter, setStatusFilter] = useState('all');
+  const [hasLoadError, setHasLoadError] = useState('');
+  const [isAdding, setIsAdding] = useState(false);
+  const [completedTodo, setCompletedTodo] = useState(false);
+  const [toggleAll, setToggleAll] = useState(false);
 
   useEffect(() => {
-    // focus the element with `ref={newTodoField}`
     if (newTodoField.current) {
       newTodoField.current.focus();
     }
+
+    if (user) {
+      setIsAdding(true);
+      getTodos(user.id)
+        .then(todo => {
+          setTodos(todo);
+        }).catch(() => (
+          setHasLoadError('Unable to load a todo')
+        )).finally(() => setIsAdding(false));
+    }
   }, []);
 
+  const handleAddTodo = async (event: FormEvent) => {
+    event.preventDefault();
+    setHasLoadError('');
+    setIsAdding(true);
+    if (user && newTitleTodo.trim() !== '') {
+      await createTodo(user.id, newTitleTodo)
+        .then(todo => {
+          setTodos([...todos, todo]);
+        })
+        .catch(() => setHasLoadError('Unable to add a todo'))
+        .finally(() => setIsAdding(false));
+    } else {
+      setHasLoadError('Title can\'t be empty');
+      setIsAdding(false);
+    }
+
+    setNewTitleTodo('');
+  };
+
+  const handleDeleteTodo = async (event: FormEvent, curentTodoId: number) => {
+    event.preventDefault();
+    setTodoId(curentTodoId);
+    setIsAdding(true);
+
+    await deleteTodos(curentTodoId)
+      .then(() => {
+        setTodos([...todos.filter(({ id }) => id !== curentTodoId)]);
+      })
+      .catch(() => setHasLoadError('Unable to delete a todo'))
+      .finally(() => setIsAdding(false));
+  };
+
+  const handleClearCompleted = () => {
+    setIsAdding(true);
+    const clearCompleted = () => ([...todos].forEach(todo => {
+      if (todo.completed === true) {
+        deleteTodos(todo.id)
+          .then(() => {
+            setTodos([...todos.filter(({ completed }) => completed !== true)]);
+          })
+          .catch(() => setHasLoadError('Unable to delete a todo'))
+          .finally(() => setIsAdding(false));
+      }
+    })
+    );
+
+    clearCompleted();
+  };
+
+  const handleChangeCompleted = async (curentTodoId: number) => {
+    setCompletedTodo(!completedTodo);
+
+    const filterTodo = todos.map(todo => {
+      if (todo.id === curentTodoId) {
+        todo.completed = completedTodo;
+      }
+
+      return todo;
+    });
+
+    await updateTodoCompleted(curentTodoId, completedTodo)
+      .then(() => {
+        setTodos(filterTodo);
+      })
+      .catch(() => setHasLoadError('Unable to completed a todo'));
+  };
+
+  const handleToggleAll = () => {
```

Comment:
When I click to button ToggleAll for the first time - all completed todos became not completed. But it shoud work another way: your active todos should became completed.

Reply:
Surprisingly! But I checked and it works correctly.🤔

Reply:
fixed it )